### DISCUSSION TITLE
How to pass the updated edge_attr to the next layer?

### DISCUSSION THREAD

**Discussion by Bowen-n** [pyg-team/pytorch_geometric#5632] [Q&A]
It seems that `self.aggregate` only returns the updated node features.

I want to update my `edge_attr` and pass is to the next layer. How to implement my `MessagePassing` model ?

**Comment by Bowen-n** [comment]
For example, how to implement GatedGCN-E mentioned in paper [Benchmarking Graph Neural Networks](https://arxiv.org/abs/2003.00982).

**Reply by rusty1s** [reply]
If you want to also update your edge features and return it as part of your GNN layer, take a look at `edge_updater` functionality within `MessagePassing`, see [here](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.MessagePassing.edge_updater). [Here](https://github.com/pyg-team/pytorch_geometric/blob/master/test/nn/conv/test_message_passing.py#L243-L259) is an example on how to use.
